### PR TITLE
Fix broken benchmarks

### PR DESCRIPTION
**Pull request by rynowak**
The tests for matching still compile but do the wrong thing, because
they aren't setting up the HTTP method metadata correctly.

Adding back an overload of CreateEndpoint that's like what was there
before.

### CODE REVIEW

**Comment by rynowak** [comment]
I thought we had that, so it was much to my chagrin that I discovered it was broken. If you know what we need to do to enable this please share it :)

**Comment by rynowak** [comment]
Ah, I figured out what was missing. On it.

**Comment by JamesNK** [comment]
How was it breaking? If it was throwing an error then I believe you can check for this during CI. The SignalR build ran the micro-benchmarks once to check that there wasn't an error.

If it is wrong logic producing an incorrect result then assert the result in the benchmark test (unless you know of a better way).

**Comment by rynowak** [comment]
I'd done the needful. We were missing some MSBuild junk to make the benchmark validation actually run. The vast majority of our benchmarks do validate their output and throw already.

File: benchmarks/Microsoft.AspNetCore.Routing.Performance/Matching/MatcherFindCandidateSetAzureBenchmark.cs [hunk 1]
```diff
@@ -10,8 +10,8 @@ namespace Microsoft.AspNetCore.Routing.Matching
     public partial class MatcherFindCandidateSetAzureBenchmark : EndpointRoutingBenchmarkBase
     {
         // SegmentCount should be max-segments + 1, but we don't have a good way to compute
-        // it here, so using 16 as a safe guess.
-        private const int SegmentCount = 16;
+        // it here, so using 32 as a safe guess.
+        private const int SegmentCount = 32;
```
Comment (rynowak) [inline benchmarks/Microsoft.AspNetCore.Routing.Performance/Matching/MatcherFindCandidateSetAzureBenchmark.cs]: Oops. This was not a safe guess.

**Comment by rynowak** [comment]
@JamesNK updated

**Pull request merged**